mx.c

```diff
@@ -1168,6 +1168,11 @@ struct Message *mx_msg_open(struct Mailbox *m, int msgno)
     return NULL;
   }
 
+  if (!(0 <= msgno < m->msg_count))
```

Comment:
Ah, I _did_ suggest that, but it's not valid C.
You'll have check the two conditions separately.

Reply:
sorry... habit from other languages where this is perfectly valid code...